packages/search/src/style.scss

```
@@ -227,6 +227,23 @@ $input-z-index: 20;
 
 			padding-left: 8px;
 		}
+
+		// Removes the fade overlay when the search is open and focused
+		&.has-focus {
+			.search-component__input-fade {
+
+				&::before {
+					display: none;
+				}
+
+				&.ltr {
```

`ltr` is the left-to-right direction and it's the default. We're overriding for `rtl`.

```suggestion
				&.rtl {
```

---

My bad. It stopped working because I added a testing change, and forgot to remove it. It's working now.

Although I'm not sure if this code covers the `.rtl` case yet. If you see the old implementation ([here](https://github.com/Automattic/wp-calypso/blob/c13b05495b71df64ce1cff2199b7fefb632deaf5/packages/search/src/style.scss#L210-L226)) It only covers the `.ltr` case. Hence I replicated the same for `.has-focus` selector.

Replacing `.ltr` with `.rtl` might introduce breaking changes.

---

Sorry, I don't understand that. LTR is the component's default and makes the input fade out from left to right at the end of it.

When the user chooses RTL we should flip where the input ends (on the left in this case), then apply the fade out from right to left. I don't get how it's a breaking change.

If you try the suggested changes in Storybook, you'll see the effect I'm describing in the "default value" and "RTL mode" story, respectively. Am I missing something?

---

I have added your suggestion. Could you please test it for both cases, when input has `.ltr` and `.rtl` direction.

You could do this by adding default `dir` [here](https://github.com/Automattic/wp-calypso/blob/b9159b553592d301d9c64685403fabf834f55abb/packages/search/src/search.tsx#L136)

---

You can also test by opening the "RTL Mode" story, there's no need to tweak the props anymore 👍

---

I can't find this story anywhere. Could you please point me towards that?

Also, I checked `rtl` case by changing the prop, it's not working properly. Shouldn't be the overlay appear on left of input box in `rtl` case?

---

You need to rebase Calypso's `trunk` to your branch 👍

---

Oh alright, now everything makes sense. Thanks :)